mmv1/products/kms/terraform.yaml

```diff
@@ -98,6 +98,23 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       decoder: templates/terraform/decoders/kms.go.erb
       encoder: templates/terraform/encoders/kms_crypto_key.go.erb
       update_encoder: templates/terraform/update_encoder/kms_crypto_key.go.erb
+  CryptoKeyVersion: !ruby/object:Overrides::Terraform::ResourceOverride
+    description: |
+      {{description}}
+
+      Destroying a cryptoKeyVersion will not delete the resource from the project.
+    id_format: "{{name}}"
+    import_format: ["{{crypto_key}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "kms_crypto_key_version_basic"
+        primary_resource_id: "example-key"
+        skip_test: true
+    properties:
+      state: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      custom_delete: templates/terraform/custom_delete/kms_crypto_key.erb
```

Comment:
This custom delete is specific to KMS crypto key - specifically, it calls `parseKmsCryptoKeyId`, which doesn't work since the resource is a kms crypto key version. (This is where the error message of "Invalid CryptoKey id format" is coming from.)

I think that since crypto key version has a [destroy](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys.cryptoKeyVersions/destroy) method, you can use that instead of having a custom delete:
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource.rb#L73

in api.yaml:

```
destroy_url: {{name}}:destroy
```

Reply:
Sure will do

Reply:
That makes sense - in that case you'll need a custom delete that succeeds if the resource turns out to already be in a destroyed state.

Reply:
Done as custom delete is set.
The destroy URL is tested in all 3 added tests, all works fine.